### PR TITLE
Fix #15705.

### LINKED ISSUE #15705: Diff - and + should be behind a setting
The `-` and `+` now in the diff view take away horizontal space, forcing the line number gutter to be squeezed. This IMO was a bad compromise.

But in any case, give me a setting to not show the `-` and `+`.

### PR DESCRIPTION
**Pull request by rebornix**
Control if +/- should be shown by a new setting `diffEditor.renderIndicators`.

### CODE REVIEW

File: src/vs/editor/browser/widget/diffEditorWidget.ts [hunk 1]
```diff
@@ -1787,12 +1811,14 @@ class InlineViewZonesComputer extends ViewZonesComputer {
 	private originalModel: editorCommon.IModel;
 	private modifiedEditorConfiguration: editorCommon.InternalEditorOptions;
 	private modifiedEditorTabSize: number;
+	private renderIndicator: boolean;
 
-	constructor(lineChanges: editorCommon.ILineChange[], originalForeignVZ: editorCommon.IEditorWhitespace[], modifiedForeignVZ: editorCommon.IEditorWhitespace[], originalEditor: editorBrowser.ICodeEditor, modifiedEditor: editorBrowser.ICodeEditor) {
+	constructor(lineChanges: editorCommon.ILineChange[], originalForeignVZ: editorCommon.IEditorWhitespace[], modifiedForeignVZ: editorCommon.IEditorWhitespace[], originalEditor: editorBrowser.ICodeEditor, modifiedEditor: editorBrowser.ICodeEditor, renderIndicator?: boolean) {
```
Comment (alexandrudima) [inline src/vs/editor/browser/widget/diffEditorWidget.ts]: same question

File: src/vs/editor/browser/widget/diffEditorWidget.ts [hunk 2]
```diff
@@ -1136,15 +1160,15 @@ class DiffEditorWidgetStyle {
 		};
 	}
 
-	_getViewZones(lineChanges: editorCommon.ILineChange[], originalForeignVZ: editorCommon.IEditorWhitespace[], modifiedForeignVZ: editorCommon.IEditorWhitespace[], originalEditor: editorBrowser.ICodeEditor, modifiedEditor: editorBrowser.ICodeEditor): IEditorsZones {
+	_getViewZones(lineChanges: editorCommon.ILineChange[], originalForeignVZ: editorCommon.IEditorWhitespace[], modifiedForeignVZ: editorCommon.IEditorWhitespace[], originalEditor: editorBrowser.ICodeEditor, modifiedEditor: editorBrowser.ICodeEditor, renderIndicators?: boolean): IEditorsZones {
```
Comment (alexandrudima) [inline src/vs/editor/browser/widget/diffEditorWidget.ts]: @rebornix why is renderIndicators optional here?

File: src/vs/editor/browser/widget/diffEditorWidget.ts [hunk 3]
```diff
@@ -1671,12 +1695,12 @@ class DiffEdtorWidgetInline extends DiffEditorWidgetStyle implements IDiffEditor
 		// Nothing to do..
 	}
 
-	_getViewZones(lineChanges: editorCommon.ILineChange[], originalForeignVZ: editorCommon.IEditorWhitespace[], modifiedForeignVZ: editorCommon.IEditorWhitespace[], originalEditor: editorBrowser.ICodeEditor, modifiedEditor: editorBrowser.ICodeEditor): IEditorsZones {
-		let computer = new InlineViewZonesComputer(lineChanges, originalForeignVZ, modifiedForeignVZ, originalEditor, modifiedEditor);
+	_getViewZones(lineChanges: editorCommon.ILineChange[], originalForeignVZ: editorCommon.IEditorWhitespace[], modifiedForeignVZ: editorCommon.IEditorWhitespace[], originalEditor: editorBrowser.ICodeEditor, modifiedEditor: editorBrowser.ICodeEditor, renderIndicators?: boolean): IEditorsZones {
```
Comment (alexandrudima) [inline src/vs/editor/browser/widget/diffEditorWidget.ts]: same question

**Review by alexdima — CHANGES_REQUESTED** [review]
LGTM, except I would remove the `?` -- i.e. IMHO adds unnecessary complexity on reading the code, since that argument is needed and used...

**Comment by rebornix** [comment]
Comments addressed :)

**Comment by rebornix** [comment]
Remove the redundant `:`, now we are good to go.

**Comment by rebornix** [comment]
Merge this first as last time when @alexandrudima reviewed this code the only comment is removing the typo.

**Pull request merged**